client/asset/eth/eth.go

```diff
@@ -3074,6 +3376,291 @@ func (w *ETHWallet) EstimateSendTxFee(addr string, value, _ uint64, subtract boo
 	return maxFee, w.ValidateAddress(addr), nil
 }
 
+// MakeBondTx creates an unsigned transaction to create a new bond. ETH
+// transactions cannot be signed beforehand due to nonce ordering, so the
+// the SignedTx and UnsignedTx fields in the return value both contain
+// the same data. The nonce and the signature is added when SendTransaction
+// is called.
+func (w *ETHWallet) MakeBondTx(ver uint16, amt uint64, feeRate uint64, lockTime time.Time, privKey *secp256k1.PrivateKey, acctID []byte) (*asset.Bond, func(), error) {
+	if ver != 0 {
+		return nil, nil, fmt.Errorf("only version 0 bonds are supported")
+	}
+
+	var accountID, bondID [32]byte
+	copy(accountID[:], acctID)
+	copy(bondID[:], privKey.Serialize())
```

Comment:
> should use the pubkey hash like utxo coins? This private key can be read by others on chain right? This leaks private info from core. Unsure if this is a problem for account bonds but just to be safe...

> I've forgotten if the server needs this for verification. For the current contract this is the key, but I think I suggested that just be a nonce. It could be any random number though correct? This passed value doesn't have to be used at all?

I don't think there's any security risk, but can't hurt to hash it. And yes, it can be just any random number. If we do a nonce, then each account has to maintain a nonce counter.. I don't see the point of that. Might as well just pass random numbers. 

> For utxo coins it is used so that we can refund the bond tx with any wallet. So the user could make an internal wallet and bond and then switch to an rpc or electrum wallet and refund that bond. Are eth users able to use different wallets like that with the current code? I guess currently the account is always imported into the new wallet so it is never a problem? There can only be the wallet from the internal seed? Maybe worth noting this distinction somewhere so we don't break eth bonds accidentally...

Yes, you will have to refund to the same address. It's 30k gas to do an additional ECDSA verification, so I don't think that's worth it to allow refunding to other wallets. As long as they don't lose the dexc seed, they'll be fine.

Reply:
> It's 30k gas to do an additional ECDSA verification, so I don't think that's worth it to allow refunding to other wallets. 

I won't push it because I don't know if we will need it but you could make another function like`refundToAnyAddress` that checks they can sign some info using the bond key and then sends the refund to any specified address. Then the normal path can stay cheap.